src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -4069,8 +4076,21 @@ internal uint GetValEscape(BoundExpression expr, uint scopeOfTheContainingExpres
                             isRefEscape: false);
                     }
 
-                    return Math.Max(GetValEscape(binary.Left, scopeOfTheContainingExpression),
-                                    GetValEscape(binary.Right, scopeOfTheContainingExpression));
+#if DEBUG
+                    // Nested binary operators on the left side are not visited by BoundTreeWalkerWithStackGuardWithoutRecursionOnTheLeftOfBinaryOperator.
+                    var assertNestedVisited = binary.Left is not BoundBinaryOperator;
+#endif
+
+                    return Math.Max(GetValEscape(binary.Left, scopeOfTheContainingExpression
+#if DEBUG
+                                        , assertNestedVisited
+#endif
+                                    ),
+                                    GetValEscape(binary.Right, scopeOfTheContainingExpression
+#if DEBUG
+                                        , assertNestedVisited
```

Comment:
>assertNestedVisited [](http://example.com/codeflow?start=42&length=19)

This looks suspicious. Why is it appropriate to pass this flag for the right hand side?

Reply:
It's not necessary to pass it for the right-hand side, thanks.

Reply:
> It's not necessary to pass it for the right-hand side

Not necessary or wrong?

Reply:
Wrong in the sense that we would not check that the right-hand side was visited.